golangci-lint failure needs to be addressed:
```
e2e/pod.go:220:33: `execCommandInContainer` - `cn` always receives `"csi-rbdplugin"` (unparam)
	f *framework.Framework, c, ns, cn string, opt *metav1.ListOptions) (string, string, error) {
	                               ^
```

--- a/internal/util/crypto.go
+++ b/internal/util/crypto.go
@@ -294,7 +294,8 @@ func DeviceEncryptionStatus(ctx context.Context, devicePath string) (mappedDevic
 	if len(lines) < 1 {
 		return "", "", fmt.Errorf("device encryption status returned no stdout for %s", devicePath)
 	}
-	if !strings.HasSuffix(lines[0], " is active.") {
+	// The line will look like: "/dev/mapper/xxx is active and is in use."
+	if !strings.Contains(lines[0], " is active") {
 		// Implies this is not a LUKS device
 		return devicePath, "", nil
 	}

--- a/internal/util/crypto.go
+++ b/internal/util/crypto.go
@@ -261,6 +261,17 @@ func OpenEncryptedVolume(ctx context.Context, devicePath, mapperFile, passphrase
 	return err
 }
 
+// ResizeEncryptedVolume resizes encrypted volume so that it can be used by the client.
+func ResizeEncryptedVolume(ctx context.Context, mapperFile string) error {
+	DebugLog(ctx, "Resizing LUKS device %s", mapperFile)
+	_, stderr, err := LuksResize(mapperFile)
+	if err != nil {
+		ErrorLog(ctx, "failed to resize LUKS device %s: %s", mapperFile, stderr)
+	}
+
+	return err
+}
+
 // CloseEncryptedVolume closes encrypted volume so it can be detached.
 func CloseEncryptedVolume(ctx context.Context, mapperFile string) error {
 	DebugLog(ctx, "Closing LUKS device %s", mapperFile)

--- a/internal/util/crypto.go
+++ b/internal/util/crypto.go
@@ -255,7 +255,7 @@ func OpenEncryptedVolume(ctx context.Context, devicePath, mapperFile, passphrase
 	DebugLog(ctx, "Opening device %s with LUKS on %s", devicePath, mapperFile)
 	_, stderr, err := LuksOpen(devicePath, mapperFile, passphrase)
 	if err != nil {
-		WarningLog(ctx, "failed to open LUKS device %q: %s", devicePath, stderr)
+		ErrorLog(ctx, "failed to open LUKS device %q: %s", devicePath, stderr)
 	}
 
 	return err

--- a/internal/util/cryptsetup.go
+++ b/internal/util/cryptsetup.go
@@ -43,6 +43,11 @@ func LuksOpen(devicePath, mapperFile, passphrase string) (stdout, stderr []byte,
 	return execCryptsetupCommand(&passphrase, "luksOpen", devicePath, mapperFile, "-d", "/dev/stdin")
 }
 
+// LuksResize resizes LUKS encrypted partition.
+func LuksResize(mapperFile string) (stdout, stderr []byte, err error) {
+	return execCryptsetupCommand(nil, "resize", mapperFile)
+}
+
 // LuksClose removes existing mapping.
 func LuksClose(mapperFile string) (stdout, stderr []byte, err error) {
 	return execCryptsetupCommand(nil, "luksClose", mapperFile)

